shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -143,6 +167,40 @@ def _add_proxy_to_graph(self, region_graph: RegionGraph):
             kwargs={},
         )
 
+    def update_arg(self, idx_or_name: int | str, value: Any):
+        """
+        Update the value of an argument in the node while keeping the
+        underlying fx.Node consistent.
+        """
+        dataclass_fields = fields(self)[5:]
+        if isinstance(idx_or_name, str):
+            if idx_or_name not in [field.name for field in dataclass_fields]:
+                raise ValueError(f"Field {idx_or_name} not found")
+            idx = [field.name for field in dataclass_fields].index(idx_or_name)
+        else:
+            idx = idx_or_name
+        if isinstance(value, CustomOp):
+            value = value.fx_node
+        # Skip the fields defined by the abstract base class
+        if 0 <= idx < len(dataclass_fields):
+            self.node_args[idx] = value  # TODO: Does this make sense?
+            field_name = dataclass_fields[idx].name
+            # Set the new value for the field
+            setattr(self, field_name, value)
+            fx_val = value.fx_node if isinstance(value, CustomOp) else value
```

Comment:
In line 182, in the instance of a CustomOp, value is set to value.fx_node. Isn't that going to be a problem here because value.fx_node.fx_node is not defined?

Reply:
Ah, I check whether value is a CustomOp twice.
This was no problem because at the second instance `isinstance(value, CustomOp)` will always evaluate to false because of line 182, where we set `value = value.fx_node` in that case. 
I removed the second check